src/Validator.php

```diff
@@ -95,19 +87,21 @@ public function rules(array $hash): self
     /**
      * Set conditional rules.
      *
-     * @param array<string, int|string>                                                                              $conditions
-     * @param array<string, array<string|array<string|int|\Closure(string, mixed, list<mixed>, list<mixed>): bool>>> $then_hash
-     * @param array<string, array<string|array<string|int|\Closure(string, mixed, list<mixed>, list<mixed>): bool>>> $else_hash
+     * @param array<string, mixed>|list<array<string, mixed>> $conditions
```

Comment:
cond type - this will be otherwise simplified by PHPStan - https://phpstan.org/r/5f05fe83-9d40-41e9-85fe-7656baa080d9

Reply:
Actually I just noticed that these types are wrong and we do not actually support them like that. But anyway will make separate phpstan-type for them to ease improving them in future.